crates/codegen/parser/runtime/src/parse_error.rs

```diff
@@ -20,12 +20,11 @@ impl ParseError {
         let tokens_that_would_have_allowed_more_progress = self
             .tokens_that_would_have_allowed_more_progress
             .iter()
-            .map(|kind| kind.as_ref())
             .collect::<BTreeSet<_>>();
 
         tokens_that_would_have_allowed_more_progress
             .into_iter()
-            .map(|token| token.to_string())
+            .map(ToString::to_string)
```

Comment:
I noticed some of these suggestions are to use the trait name `ToString::to_string` and the other use the concrete type name `TokenKind::to_string`. Not sure why.

Reply:
Do you mean unstable suggestions, i.e. Clippy sometimes suggests trait name and sometimes the Self type name?

For associated functions like `PrattElement::into_nodes` the type name is suggested; when a trait implementation is used in the closure, the trait name is used (e.g. `ToString::to_string`).

In this case it's clear to indicate the trait name, as we know it's a standard trait (e.g. Display impl) and not a custom function but that's not always the case - for instance, it's clearer to use the type name when calling `Default::default`. As always, it depends! Do you have a specific preference here?

Reply:
Like I said, it probably greatly depends on the call-site but using the concrete receiver type is a good default, agreed. Done in https://github.com/NomicFoundation/slang/pull/695/commits/25b624b790a35701cf6b705a855734657566f477